src/shape/point.js

```diff
@@ -2,32 +2,121 @@
  * Copyright (c) 2017 NAVER Corp.
  * billboard.js project is licensed under the MIT license
  */
+import {select as d3Select} from "d3";
 import ChartInternal from "../internals/ChartInternal";
-import {isFunction, isObjectType, extend} from "../internals/util";
+import {isFunction, isObjectType, extend, notEmpty} from "../internals/util";
 
 extend(ChartInternal.prototype, {
 	hasValidPointType(type) {
 		return /^(circle|rect(angle)?|polygon|ellipse)$/i.test(type || this.config.point_type);
 	},
 
-	hasValidPointDrawMethods() {
-		const pointType = this.config.point_type;
+	hasValidPointDrawMethods(type) {
+		const pointType = type || this.config.point_type;
 
 		return isObjectType(pointType) &&
 			isFunction(pointType.create) && isFunction(pointType.update);
 	},
 
+	insertPointInfoDefs(point, id) {
+		const $$ = this;
+		const defs = $$.svg.select("defs");
+
+		if (defs.size < 1) {
+			return;
+		}
+
+		const html = defs.html();
+
+		// Add the point node into <defs>
+		defs.html(`${html}${point}`);
+
+		const node = defs.node().lastChild;
+
+		node.setAttribute("id", id);
```

Comment:
When set the exact ID value, there's a possibility having some collision if there're more than one chart instance in same page.

![image](https://user-images.githubusercontent.com/2178435/33924993-b763ceae-e01b-11e7-8a0d-7369c4f1851c.png)

As you can see the node constructed, others having ID with prefixed `bb-${datetime}` format.
You can prefix with the `$$.clipId`.
https://github.com/naver/billboard.js/blob/9f2eb7ed0511cbe57d722a863948d45379483256/src/internals/ChartInternal.js#L85

```js
$$.dtPrefix = `bb-${+new Date()}`;
$$.clipId = `${dtPrefix}-clip`;
```

Reply:
@netil thanks! I'll use this instead